tst/EnergyPlus/api/TestDataTransfer.py

```diff
@@ -57,31 +57,63 @@
 from pyenergyplus.api import EnergyPlusAPI
 
 one_time = True
-outdoor_temp_sensor = 0
-outdoor_dew_point_sensor = 0
-outdoor_dew_point_actuator = 0
+outdoor_temp_sensor = -1
+outdoor_dew_point_sensor = -1
+outdoor_dew_point_actuator = -1
+zone1_wall1_actuator_handle = -1
+wall_construction_handle = -1
+floor_construction_handle = -1
+ok = True
+exception = None
 
 
 def time_step_handler(state):
-    global one_time, outdoor_temp_sensor, outdoor_dew_point_sensor, outdoor_dew_point_actuator
+    global one_time, outdoor_temp_sensor, outdoor_dew_point_sensor, outdoor_dew_point_actuator, zone1_wall1_actuator_handle, wall_construction_handle, floor_construction_handle, ok, exception
     sys.stdout.flush()
+    # If API isn't ready yet, or we know it failed already, just return
+    if not api.exchange.api_data_fully_ready(state) or not ok:
+        return
+
+    # Get the handles
     if one_time:
-        if api.exchange.api_data_fully_ready(state):
+        one_time = False
+        try:
             # val = api.exchange.list_available_api_data_csv()
             # with open('/tmp/data.csv', 'w') as f:
             #     f.write(val.decode(encoding='utf-8'))
             outdoor_temp_sensor = api.exchange.get_variable_handle(
-                state, u"SITE OUTDOOR AIR DRYBULB TEMPERATURE", u"ENVIRONMENT"
+                state, "SITE OUTDOOR AIR DRYBULB TEMPERATURE", "ENVIRONMENT"
             )
             outdoor_dew_point_sensor = api.exchange.get_variable_handle(
-                state, u"SITE OUTDOOR AIR DEWPOINT TEMPERATURE", u"ENVIRONMENT"
+                state, "SITE OUTDOOR AIR DEWPOINT TEMPERATURE", "ENVIRONMENT"
             )
             outdoor_dew_point_actuator = api.exchange.get_actuator_handle(
                 state, "Weather Data", "Outdoor Dew Point", "Environment"
             )
-            if outdoor_temp_sensor == -1 or outdoor_dew_point_sensor == -1 or outdoor_dew_point_actuator == -1:
-                sys.exit(1)
-            one_time = False
+            zone1_wall1_actuator_handle = api.exchange.get_actuator_handle(
+                state, "Surface", "Construction State", "Zn001:Wall001"
+            )
+
+            wall_construction_handle = api.exchange.get_construction_handle(state, "R13WALL")
+            floor_construction_handle = api.exchange.get_construction_handle(state, "FLOOR")
+        except Exception as e:
+            # Capture ok and exception message
+            exception = e
+            ok = False
+            raise e
+
+        # If the one_time block failed before, E+ swallowed the exception, but our handles will be wrong
+        if (
+            outdoor_temp_sensor == -1
+            or outdoor_dew_point_sensor == -1
+            or outdoor_dew_point_actuator == -1
+            or zone1_wall1_actuator_handle == -1
+            or wall_construction_handle == -1
+            or floor_construction_handle == -1
+        ):
+            ok = False
+            return
```

Comment:
changing the control flow + getting some construction handles

Reply:
This is fine but a bit funny looking.  I think it would be nicer as:

```python
handles = [
    outdoor_temp_sensor,
    outdoor_dew_point_sensor,
    outdoor_dew_point_actuator,
    zone1_wall1_actuator_handle,
    wall_construction_handle,
    floor_construction_handle
]
if -1 in handles:
    ...
```

No need to change it here, just pointing out my preferred style.